apigw/src/shared/saml/passport-saml-cache-redis.ts

```diff
@@ -33,41 +32,35 @@ export default function redisCacheProvider(
   client: RedisClient,
   options: ProviderOptions
 ): CacheProvider {
-  const { ttlSeconds = 60 * 60, keyPrefix } = { ...options }
+  const { ttlSeconds = 60 * 60, keyPrefix } = options
 
   if (!Number.isInteger(ttlSeconds) || ttlSeconds <= 0) {
     throw new Error('ttlSeconds must be a positive integer')
   }
 
   return {
-    getAsync: (key) =>
-      fromCallback((callback) => client.get(`${keyPrefix}${key}`, callback)),
-    saveAsync: (key, value) =>
-      fromCallback((callback) =>
-        client.get(`${keyPrefix}${key}`, (err, reply) => {
-          if (err) return callback(err, null)
-          if (reply !== null) return callback(null, null)
+    getAsync: (key: string) => client.get(`${keyPrefix}${key}`),
+    saveAsync: async (key: string, value: string) => {
+      const reply = await client.get(`${keyPrefix}${key}`)
+      if (reply !== null) return null
 
-          const cacheItem: CacheItem = {
-            createdAt: new Date().getTime(),
-            value
-          }
-          return client.set(
-            `${keyPrefix}${key}`,
-            value,
-            'EX',
-            ttlSeconds,
-            (err) => callback(err, cacheItem)
-          )
+      const cacheItem: CacheItem = {
+        createdAt: new Date().getTime(),
+        value
+      }
+      try {
+        await client.set(`${keyPrefix}${key}`, value, {
+          EX: ttlSeconds
         })
-      ),
-    removeAsync: (key) =>
-      fromCallback((callback) =>
-        client.del(`${keyPrefix}${key}`, (err, count) => {
-          if (err) return callback(err, null)
-          if (count === 0) return callback(null, null)
-          return callback(null, key)
-        })
-      )
+      } catch (err: unknown) {
```

Comment:
Why is the error caught and discarded here? :thinking:

Reply:
Seems I misunderstood the callback in the original code. Fixed.